website/source/guides/secret-mgmt/db-root-rotation.html.md

```diff
@@ -0,0 +1,409 @@
+---
+layout: "guides"
+page_title: "DB Root Credential Rotation - Guides"
+sidebar_current: "guides-secret-db-root-rotation"
+description: |-
+  Vault enables the combined database secret engines to automate the rotation of
+  root credentials.
+---
+
+# Database Root Credential Rotation
+
+## Database Secrets Engine
+
+Vault's [database secrets engine](/docs/secrets/databases/index.html) is used to
+dynamically generate role-based database credentials for various database
+systems. The generation of dynamic database credentials allows every service
+instance to have a unique set of database credentials instead of sharing one.
+Since the credentials are tied directly to each service instance and live only
+for the life of the service, any abnormal access pattern can be mapped to a
+specific service instance and its credential can be revoked immediately.
+
+This reduced the manual tasks performed by the database administrator and made
+the access to the database to be more efficient and secure.
+
+The [Secret as a Service: Dynamic Secrets](/guides/secret-mgmt/dynamic-secrets.html)
+guide demonstrates the basic working of it.
+
+## Reference Material
+
+- [Database Secret Engine (API)](/api/secret/databases/index.html#rotate-root-credentials)
+- [PostgreSQL Database Plugin HTTP API](/api/secret/databases/postgresql.html)
+- [Secret as a Service: Dynamic Secrets](/guides/secret-mgmt/dynamic-secrets.html)
+
+## Estimated Time to Complete
+
+10 minutes
+
+
+## Challenge
+
+Because Vault is managing credentials on behalf of the database administrator,
+it must also be given a set of highly privileged credentials which can grant and
+revoke access to the database system.  Therefore, it is very common to give
+Vault the **root** credentials.
+
+However, these credentials are often long-lived and never change once configured
+on Vault. This may violate the _Governance, Risk and Compliance_ (GRC)
+surrounding that data stored in the database.
+
+
+## Solution
+
+Use the Vault's **`/database/rotate-root/:name`** API endpoint to rotate the
+root credentials stored for the database connection.
+
+![DB Root Credentials](/assets/images/vault-db-root-rotation.png)
+
+## Prerequisites
+
+To perform the tasks described in this guide, you need to have a Vault
+environment.  Refer to the [Getting
+Started](/intro/getting-started/install.html) guide to install Vault. Make sure
+that your Vault server has been [initialized and
+unsealed](/intro/getting-started/deploy.html).
+
+### PostgreSQL
+
+This guide requires that you have a PostgreSQL server to connect to. If you
+don't have one, install [PostgreSQL](https://www.postgresql.org/download/) to
+perform the steps described in this guide.
+
+**Example on CentOS:**
+
+```shell
+# To list available packages
+$ yum list postgreql*
```

Comment:
I'm not sure we want to provide install instructions for postgres here.  Maybe we can link to a procedure from the official documents.

Reply:
Easy fix!  I'll do that. :)